tools-v2/pkg/cli/command/curvebs/snapshot/copyset/all.go

```diff
@@ -0,0 +1,161 @@
+	return NewSnapshotAllCopysetCommand().Cmd
+}
+func (sCmd *SnapshotAllCopysetCommand) AddFlags() {
+	config.AddRpcRetryTimesFlag(sCmd.Cmd)
+	config.AddRpcTimeoutFlag(sCmd.Cmd)
+}
+
+func (sCmd *SnapshotAllCopysetCommand) Init(cmd *cobra.Command, args []string) error {
+	timeout := config.GetFlagDuration(sCmd.Cmd, config.RPCTIMEOUT)
+	retryTimes := config.GetFlagInt32(sCmd.Cmd, config.RPCRETRYTIMES)
+
+	chunkserverInfos, err := chunkserver.GetChunkServerInCluster(sCmd.Cmd)
+	if err.TypeCode() != cmderror.CODE_SUCCESS {
+		return err.ToError()
+	}
+
+	csAddrs := make([]string, len(chunkserverInfos))
+	for i, info := range chunkserverInfos {
+		csAddrs[i] = fmt.Sprintf("%s:%d", info.GetHostIp(), info.GetPort())
+	}
+
+	sCmd.SetHeader([]string{cobrautil.ROW_CHUNKSERVER, cobrautil.ROW_RESULT})
+	sCmd.TableNew.SetAutoMergeCellsByColumnIndex(cobrautil.GetIndexSlice(
+		sCmd.Header, []string{cobrautil.ROW_RESULT},
+	))
+
+	sCmd.rows = make([]map[string]string, len(csAddrs))
+	for i, addr := range csAddrs {
+		sCmd.rows[i] = make(map[string]string)
+		sCmd.rows[i][cobrautil.ROW_CHUNKSERVER] = addr
+		rpc := &SnapshotAllRpc{
+			Request: &cli2.SnapshotAllRequest{},
+			Info:    basecmd.NewRpc([]string{addr}, timeout, retryTimes, "SnapshotAll"),
+		}
+		sCmd.rpcs = append(sCmd.rpcs, rpc)
+	}
+
+	return nil
+}
+
+func (sCmd *SnapshotAllCopysetCommand) RunCommand(cmd *cobra.Command, args []string) error {
+	var infos []*basecmd.Rpc
+	var funcs []basecmd.RpcFunc
+	for _, rpc := range sCmd.rpcs {
+		infos = append(infos, rpc.Info)
+		funcs = append(funcs, rpc)
+	}
+
+	results, errs := basecmd.GetRpcListResponse(infos, funcs)
+	mergeErr := cmderror.MergeCmdErrorExceptSuccess(errs)
+	sCmd.Error = mergeErr
+	if len(errs) == len(infos) {
+		return mergeErr.ToError()
+	}
+
+	for i, result := range results {
+		if result == nil {
+			sCmd.rows[i][cobrautil.ROW_RESULT] = cobrautil.ROW_VALUE_FAILED
+		} else {
+			sCmd.rows[i][cobrautil.ROW_RESULT] = cobrautil.ROW_VALUE_SUCCESS
+		}
+	}
+
+	list := cobrautil.ListMap2ListSortByKeys(sCmd.rows, sCmd.Header, []string{cobrautil.ROW_RESULT})
+	sCmd.TableNew.AppendBulk(list)
+	sCmd.Result = sCmd.rows
+	return nil
+}
+
+func (sCmd *SnapshotAllCopysetCommand) Print(cmd *cobra.Command, args []string) error {
+	return output.FinalCmdOutput(&sCmd.FinalCurveCmd, sCmd)
+}
+
+func (sCmd *SnapshotAllCopysetCommand) ResultPlainOutput() error {
+	return output.FinalCmdOutputPlain(&sCmd.FinalCurveCmd)
+}
+
+func GetAllSnapshotResult(caller *cobra.Command) (interface{}, interface{}, *cmderror.CmdError) {
+	sCmd := NewSnapshotAllCopysetCommand()
+	config.AlignFlagsValue(caller, sCmd.Cmd, []string{})
+	sCmd.Cmd.SilenceErrors = true
+	sCmd.Cmd.SilenceUsage = true
+	sCmd.Cmd.SetArgs([]string{"--format", config.FORMAT_NOOUT})
+	err := sCmd.Cmd.Execute()
+	if err != nil {
+		retErr := cmderror.ErrBsGetAllSnapshotResult()
+		retErr.Format(err.Error())
+		return nil, nil, retErr
+	}
+	return sCmd.TableNew, sCmd.Result, cmderror.Success()
```

Comment:
You can return the response (or a map) directly instead of the table, the table has no information